src/common/src/array/stream_chunk.rs

```diff
@@ -189,6 +189,9 @@ impl StreamChunk {
     }
 
     pub fn to_protobuf(&self) -> PbStreamChunk {
+        if !self.is_compacted() {
```

Comment:
Will this affect all exchange rpc and is this expected?

Reply:
I think it is a fallback logic to make sure the `to_protobuf` is true. In most of places it has been compacted.